indexing-service/src/main/java/io/druid/indexing/common/task/AbstractTask.java

```diff
@@ -98,6 +106,25 @@ public String getGroupId()
     return groupId;
   }
 
+  /**
+   * Tasks with no lock priority set will have the respective default priorities as per the task type
+   *  - Default lock priorities for task
+   *    - Realtime Index Task - 75
+   *    - Hadoop/Index Task - 50
+   *    - Merge/Append Task - 25
+   *    - Other Tasks - 0
+   * */
+  @JsonIgnore
+  @Override
+  public int getLockPriority() {
+    return getContextValue("lockPriority") == null ? DEFAULT_TASK_PRIORITY : Integer.parseInt(getContextValue("lockPriority").toString());
+  }
+
+  @JsonIgnore
+  public int getBatchSize() {
+    return getContextValue("batchSize") == null ? DEFAULT_BATCH_SIZE : Integer.parseInt(getContextValue("batchSize").toString());
```

Comment:
This should be greater than 0 or else your % operation is going to get interesting.